package/MDAnalysis/lib/distances.py

```diff
@@ -208,6 +208,23 @@ def _check_lengths_match(*arrays):
         raise ValueError("Input arrays must all be same shape"
                          "Got {0}".format([a.shape for a in arrays]))
 
+
+def distance(a, b, box=None):
+    """calc distances between two points. Uses algorithm from Tuckerman
+
+    """
+    if box is None:
+        return np.linalg.norm(a - b)
+
+    box = triclinic_vectors(box)
+    ibox = np.linalg.inv(box)
```

Comment:
Thanks. And yes, you're right, of course, the matrix-vector multiplications are bad.